### ISSUE TITLE
Widget does not work correctly in `xeus-python`

### ISSUE THREAD

**Issue by trungleduc** [jupyter-xeus/xeus-python#493]
## Description

The `dashboard.ipynb` notebook in `voila/notebooks` does not work as expected in `xeus-python` w.r.t `ipykernel`. 

- Notebook executed with  `Ipykernel` 

https://user-images.githubusercontent.com/4451292/134864505-a2086082-c08f-4d18-984c-082d1dd8246c.mp4

- Notebook executed with  `xeus-python` 

https://user-images.githubusercontent.com/4451292/134864542-eacc923f-24d0-4e09-bb81-7f47a6913eaa.mp4

The root of problem is that the [`count` attribute of `bqplot.Hist`](https://github.com/bqplot/bqplot/blob/d8fae93274422e72b7ecf1f464d8d8197103a28d/bqplot/marks.py#L876) is computed [in frontend](https://github.com/bqplot/bqplot/blob/d8fae93274422e72b7ecf1f464d8d8197103a28d/js/src/HistModel.ts#L119) then it is synchronized back to backend as a `Float64Array`.  `Ipykernel` received correctly this array but  `xeus-python` received an empty array.

- In  `Ipykernel` 

![sc_ipy](https://user-images.githubusercontent.com/4451292/134865643-8e9d118c-36e8-4b65-bb3a-258408bd9800.png)

- In  `xeus-python` 

![sc_xpy](https://user-images.githubusercontent.com/4451292/134865653-860dcbee-6baa-4758-b73b-eded924a7974.png)

## Context

```
bqplot                    0.12.30            pyhd8ed1ab_0    conda-forge
ipykernel                 6.4.1            py39hef51801_0    conda-forge
ipython                   7.27.0           py39hef51801_0    conda-forge
jupyterlab                3.1.13             pyhd8ed1ab_0    conda-forge
xeus                      2.1.0                h7d0c39e_0    conda-forge
xeus-python               0.13.0           py39h1aaad98_2    conda-forge
xeus-python-shell         0.1.5              pyhd8ed1ab_0    conda-forge
```

**Comment by julian-west** [comment]
@trungleduc -- I noticed that 'ipykernel' is hard-coded into the `comm` attribute of the Widget class. I'm wondering whether this is causing an issue for widgets trying to communicate with the xeus-python kernel.

https://github.com/jupyter-widgets/ipywidgets/blob/ab01828452f39c3e972e04408d88357f4aa1e1f4/ipywidgets/widgets/widget.py#L364

**Comment by martinRenou** [comment]
I can have a look at this one if you want @trungleduc

**Comment by trungleduc** [comment]
@martinRenou that would be great. Thank you!

**Comment by martinRenou** [comment]
> I noticed that 'ipykernel' is hard-coded into the comm attribute of the Widget class. I'm wondering whether this is causing an issue for widgets trying to communicate with the xeus-python kernel.

@julian-west this shouldn't be a problem. ipywidgets should be using xeus-python's Comms properly, otherwise no widgets would be able to display.

I believe it has to do with the way the deserialization of binary buffers work. It might be an issue in `pybind11_json`... We need to dig deeper there.

**Comment by martinRenou** [comment]
Should be fixed by #495
